### ISSUE TITLE
DISCUSSION: Make event display Angular independent

### ISSUE THREAD

**Issue by 9inpachi** [HSF/phoenix#134]
Hi,

I think this is something we should do - make the Event Display and all the required services Angular independent and create them as simple classes. The idea is to separate the Phoenix API completely from the frontend Angular application and then use the API in the application. This API will not include Angular specific components like the Phoenix menu and UI menu.

The motivation behind this is the usability and adaptability of Phoenix to other potential projects as an API.

The flow I have in mind for use in a normal experiment in our current application is:
(Highly tentative - I think we can optimize the flow further for easier usage)

```ts
// Create the configuration
const configuration = new Configuration('three-canvas-id');

// ... other possible configuration options here

// Create the event display
const eventDisplay = new EventDisplay(configuration);

// Create and set the event loader for the event display
const phoenixLoader = new PhoenixLoader();
eventDisplay.setEventLoader(phoenixLoader);

// Load the event using the event loader
// eventJSON - JSON read through some other service
// or just a URL processed by the event display internally
eventDisplay.loadEvent(eventJSON);

// Load detector geometry
eventDisplay.loadGeometry(gltfOrObjGeometryURL/*, other options */);
```

**Comment by EdwardMoyse** [comment]
I think this is a good idea! The workflow above looks good ... but how does this work with the .html experiment files?

**Comment by 9inpachi** [comment]
> how does this work with the .html experiment files

Do you mean the component templates in Angular?
It'd be pretty simple. We only need ID for the canvas (or a div we can append to) used by three.js.

And we'd also need to think how we would accommodate the UI service this way or if we should change the architecture for it.

**Comment by EdwardMoyse** [comment]
Yes, I meant the component templates. Okay - this sounds good.

I'm not sure I understand how the UI Service would work. Maybe we can chat about this this week?

**Comment by 9inpachi** [comment]
I made the UI and Three service into managers which we can access through the event display.
It works quite well - but if someone wants to adapt the functions in their own app, they will have to create their own components if they are not using Angular (that makes sense because Phoenix is a library/API).

I have a sample application with React ready to roll. I will upload it to a repo soon.

**Comment by 9inpachi** [comment]
I am facing problems with what kind of modules we can generate to be reusable. I am using "esnext" for now and it works fine with Angular and React. But I would like to generate a single UMD module which anyone can include in their HTML files through a script tag and dependencies. But haven't had much luck with it. :(

**Comment by 9inpachi** [comment]
Take a look at usage in react: https://github.com/9inpachi/phoenix-react

To make it work, we need the following steps since "@phoenix/event-display" is not on npm yet.

* Get the files from https://github.com/9inpachi/phoenix/tree/wip-eventdisplay
* In the `/packages/phoenix-event-display` directory, open terminal and type `npm link`
* Now get back to the React app and remove `@phoenix/event-display` in the `package.json` temporarily
* And then in root folder of React app with `package.json`, run the `npm link "@phoenix/event-display"` command
* And add the `@phoenix/event-display` back to `package.json`

**Comment by 9inpachi** [comment]
Closed with #140
